packages/ingest/src/granule.ts

```diff
@@ -1,71 +1,109 @@
-'use strict';
+import * as errors from '@cumulus/errors';
+import moment from 'moment';
+import * as S3 from '@cumulus/aws-client/S3';
+import * as log from '@cumulus/common/log';
+import { DuplicateHandling } from '@cumulus/types';
+
+export interface EventWithDuplicateHandling {
+  config: {
+    collection: {
+      duplicateHandling?: DuplicateHandling
+    },
+    duplicateHandling?: DuplicateHandling,
+  },
+  cumulus_config?: {
+    cumulus_context?: {
+      forceDuplicateOverwrite?: boolean
+    }
+  },
+}
 
-const encodeurl = require('encodeurl');
-const errors = require('@cumulus/errors');
-const get = require('lodash/get');
-const moment = require('moment');
-const S3 = require('@cumulus/aws-client/S3');
-const log = require('@cumulus/common/log');
-const { deprecate } = require('@cumulus/common/util');
+export interface File {
+  bucket?: string,
+  key?: string,
+  fileName?: string,
+  name?: string,
+  filename?: string
+}
+
+export interface MoveFileParams {
+  source?: {
+    Bucket: string,
+    Key: string
+  },
+  target?: {
+    Bucket: string,
+    Key: string
+  },
+  file: File
+}
 
 /**
-* Copy granule file from one s3 bucket & keypath to another
-*
-* @param {Object} source - source
-* @param {string} source.Bucket - source
-* @param {string} source.Key - source
-* @param {Object} target - target
-* @param {string} target.Bucket - target
-* @param {string} target.Key - target
-* @param {Object} options - optional object with properties as defined by AWS API:
-* https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#copyObject-property
-* @returns {Promise} returms a promise that is resolved when the file is copied
-**/
-function copyGranuleFile(source, target, options = {}) {
-  deprecate(
-    '@cumulus/ingest/granule.copyGranuleFile',
-    '1.22.1',
-    '@cumulus/aws-client/S3.multipartCopyObject'
-  );
+  * rename s3 file with timestamp
+  *
+  * @param {string} bucket - bucket of the file
+  * @param {string} key - s3 key of the file
+  * @returns {Promise} promise that resolves when file is renamed
+  */
+export async function renameS3FileWithTimestamp(
+  bucket: string,
+  key: string
+): Promise<void> {
```

Comment:
So you made the same change in the other PR and said the point of doing `await` instead of `return` was to hide the implementation? Do you mean by not exposing the actual return of `S3.moveObject`? Is there some reason that is preferable?

Reply:
Correct. That prevents people from expecting to get that exact return value back. If we need to change the way that we're implementing `renameS3FileWithTimestamp` to use some other S3 function, we won't break anybody if we aren't returning that underlying response.